examples/distributed_training_launch.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import argparse
+
+import torch.nn.functional as F
+import torchvision
+import torchvision.transforms as transforms
+from torch.optim import SGD
+
+from mmengine.dist import launch
+from mmengine.evaluator import BaseMetric
+from mmengine.model import BaseModel
+from mmengine.runner import Runner
+
+
+class MMResNet50(BaseModel):
+
+    def __init__(self):
+        super().__init__()
+        self.resnet = torchvision.models.resnet50()
+
+    def forward(self, imgs, labels, mode):
+        x = self.resnet(imgs)
+        if mode == 'loss':
+            return {'loss': F.cross_entropy(x, labels)}
+        elif mode == 'predict':
+            return x, labels
+
+
+class Accuracy(BaseMetric):
+
+    def process(self, data_batch, data_samples):
+        score, gt = data_samples
+        self.results.append({
+            'batch_size': len(gt),
+            'correct': (score.argmax(dim=1) == gt).sum().cpu(),
+        })
+
+    def compute_metrics(self, results):
+        total_correct = sum(item['correct'] for item in results)
+        total_size = sum(item['batch_size'] for item in results)
+        return dict(accuracy=100 * total_correct / total_size)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Distributed Training')
+    parser.add_argument(
+        '--launcher',
+        choices=['none', 'pytorch', 'slurm', 'mpi'],
+        default='none',
+        help='job launcher')
+    parser.add_argument(
+        '--num-gpus',
+        nargs='+',
+        type=int,
+        help='number of gpus to use',
+        default=[2])
+    parser.add_argument(
+        '--num-machines',
+        type=int,
+        help='number of machines to use',
+        default=1)
+    parser.add_argument(
+        '--machine-rank',
+        type=int,
+        help='The rank of current machine.',
+        default=0)
+    parser.add_argument(
+        '--master-addr',
+        type=str,
+        help='The FQDN of the host that is running worker with rank 0.',
```

Comment:
What's the meaning of `FQDN`?

Reply:
Just borrow the description in https://pytorch.org/docs/stable/elastic/run.html#environment-variables, I'll describe it more detail.